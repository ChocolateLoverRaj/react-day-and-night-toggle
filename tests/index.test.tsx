import React from 'react'
import renderer from 'react-test-renderer'
import DayAndNightToggler from '../src/index'


describe('Test of component <DayAndNightToggler>:', () => {
    let theme: boolean
    let changeTheme: jest.Mock
    let component: renderer.ReactTestRenderer

    beforeEach(() => {
        theme = false
        changeTheme = jest.fn(() => theme = !theme)
        renderer.act(() => {
            component = renderer.create(<DayAndNightToggler checked={theme} onChange={changeTheme} />)
        })
    })
    
    test('should render:', () => {
        expect(component.toJSON()).toMatchSnapshot()
    })
    test('should have all the necessary props:', () => { 
        expect(component.root.props).toEqual({ checked: theme, onChange: changeTheme })
        expect(changeTheme).toHaveBeenCalled
    })
    test('should all props be passed:', () => {
        renderer.act(() => {
            component.update(
                <DayAndNightToggler
                    checked={theme}
                    onChange={changeTheme}
                    size={24}
                    startInactive={true}
                    animationInactive={true}
                    shadows={true}
                    className={'class_name'}
                />
            )
        })
        expect(component.toJSON()).toMatchSnapshot()
        expect(component.root.props).toEqual({
            checked: theme,
            onChange: changeTheme,
            size: 24,
            startInactive: true,
            animationInactive: true,
            shadows: true,
            className: 'class_name'
        })
        expect(changeTheme).toHaveBeenCalled
    })
})
