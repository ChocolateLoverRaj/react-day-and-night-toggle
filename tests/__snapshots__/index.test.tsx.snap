// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of component <DayAndNightToggler>: should all props be passed: 1`] = `
<div
  className="sc-bdnxRM gKaFwy class_name"
  size={24}
>
  <div
    checked={false}
    className="sc-gtsrHT cRNVmd"
    onClick={[Function]}
    size={24}
  >
    <div
      className="sc-dlnjwi dUJAIp"
    >
      <div
        className="sc-hKFxyN fbhGat"
      />
      <div
        className="sc-eCApnc"
      >
        <div
          className="sc-jSFjdj kLwpYE"
        />
        <div
          className="sc-jSFjdj kLwpYE"
        />
        <div
          className="sc-jSFjdj kLwpYE"
        />
        <div
          className="sc-gKAaRy biuKKK"
        >
          <div
            className="sc-iCoGMd bvBrMn"
          />
        </div>
      </div>
      <div
        className="sc-pNWdM"
      >
        <div
          className="sc-jrsJWt ccGWJZ"
        >
          <div
            className="sc-fujyAs"
          />
          <div
            className="sc-fujyAs"
          />
        </div>
        <div
          className="sc-jrsJWt ccGWJZ"
        >
          <div
            className="sc-fujyAs"
          />
          <div
            className="sc-fujyAs"
          />
        </div>
        <div
          className="sc-jrsJWt ccGWJZ"
        >
          <div
            className="sc-fujyAs"
          />
          <div
            className="sc-fujyAs"
          />
          <div
            className="sc-fujyAs"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test of component <DayAndNightToggler>: should render: 1`] = `
<div
  className="sc-bdnxRM ijVOMr"
  size={32}
>
  <div
    checked={false}
    className="sc-gtsrHT Dygr"
    onClick={[Function]}
    size={32}
  >
    <div
      className="sc-dlnjwi dUJAIp"
    >
      <div
        className="sc-hKFxyN fbhGat"
      />
      <div
        className="sc-eCApnc"
      >
        <div
          className="sc-jSFjdj kLwpYE"
        />
        <div
          className="sc-jSFjdj kLwpYE"
        />
        <div
          className="sc-jSFjdj kLwpYE"
        />
        <div
          className="sc-gKAaRy biuKKK"
        >
          <div
            className="sc-iCoGMd bvBrMn"
          />
        </div>
      </div>
      <div
        className="sc-pNWdM"
      >
        <div
          className="sc-jrsJWt ccGWJZ"
        >
          <div
            className="sc-fujyAs"
          />
          <div
            className="sc-fujyAs"
          />
        </div>
        <div
          className="sc-jrsJWt ccGWJZ"
        >
          <div
            className="sc-fujyAs"
          />
          <div
            className="sc-fujyAs"
          />
        </div>
        <div
          className="sc-jrsJWt ccGWJZ"
        >
          <div
            className="sc-fujyAs"
          />
          <div
            className="sc-fujyAs"
          />
          <div
            className="sc-fujyAs"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
